### PR TITLE
CONCF-979 Add the entry point for Curated Content Tool

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/CONCF-979

In this PR we've added user rights to the `CurrentUser` and display an edit button on the main page when user has the `curatedcontent` right.

While working on it I've refactored `UserModel` a bit so maybe @mklucsarits you want to take a look?

### CODE REVIEW

**Comment by rogatty** [comment]
Currently it's missing styles for the edit button, that's why the `WIP` label.

File: front/scripts/main/CurrentUser.ts [hunk 1]
```diff
@@ -2,23 +2,78 @@
...
+	}
+}
+
 App.CurrentUser = Em.Object.extend({
-	model: null,
+	rights: {},
+	isAuthenticated: Em.computed.bool('userId'),
+
+	userId: Em.computed(function (): number {
+		var cookieUserId = ~~M.prop('userId');
```
Comment (hakubo) [inline front/scripts/main/CurrentUser.ts]: While I like usages like this parseInt is easier to understand.

File: front/scripts/main/CurrentUser.ts [hunk 2]
```diff
@@ -2,23 +2,78 @@
...
+		return cookieUserId > 0 ? cookieUserId : null;
+	}),
 
 	init: function (): void {
 		var userId = this.get('userId');
 		if (userId !== null) {
-			App.UserModel.find({userId: userId}).then((result: any): void => {
-				this.set('model', result);
-			});
+			App.UserModel.find({userId: userId})
```
Comment (hakubo) [inline front/scripts/main/CurrentUser.ts]: Please use object shorthand

File: front/scripts/main/components/ArticleWrapperComponent.ts [hunk 3]
```diff
@@ -70,6 +70,12 @@ App.ArticleWrapperComponent = Em.Component.extend(App.LanguagesMixin, App.TrackC
 		return !this.get('model.isMainPage') && this.get('isJapaneseWikia');
 	}),
 
+	curatedContentToolButtonVisible: Em.computed('model.isMainPage', 'currentUser.rights.curatedcontent',
```
Comment (hakubo) [inline front/scripts/main/components/ArticleWrapperComponent.ts]: Em.computed.and

File: front/scripts/main/CurrentUser.ts (hunk 1 above)
Comment (vforge) [inline front/scripts/main/CurrentUser.ts]: I agree

**Comment by vforge** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**